LanguageFeatures/Patterns/matching_relational_A02_t02.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion At runtime, a pattern is matched against a value. This determines
+/// whether or not the match fails and the pattern refutes the value. If the
+/// match succeeds, the pattern may also destructure data from the object or
+/// bind variables.
+///
+/// Refutable patterns usually occur in a context where match refutation causes
+/// execution to skip over the body of code where any variables bound by the
+/// pattern are in scope. If a pattern match failure occurs in an irrefutable
+/// context, a runtime error is thrown.
+///
+/// To match a pattern p against a value v:
+/// ...
+/// Relational:
+/// i. Evaluate the right-hand constant expression to c.
+/// ii. If the operator is ==:
+///   a. Let r be the result of v == c.
+///   b. The pattern matches if r is true and fails otherwise. This takes into
+///     account the built-in semantics that null is only equal to null. The
+///     result will always be a bool since operator == on Object is declared to
+///     return bool.
+/// iii. Else if the operator is !=:
+///   a. Let r be the result of v == c.
+///   b. The pattern matches if r is false and fails otherwise. This takes into
+///     account the built-in semantics that null is only equal to null.
+/// iv. Else the operator is a comparison operator op:
+///   a. Let r be the result of calling op on v with argument c.
+///   b. If r is not a bool then throw a runtime error. This can happen if the
+///     operator on v's type returns dynamic.
+///   c. The pattern matches if r is true and fails otherwise.
+///
+/// @description Check that in case `c != null` only `v == null` doesn't match.
```

Comment:
We can't actually verify that _every_ other object than null will be a non-match, but we can try it out with some specific objects. ;-)